pkg/codegen/testing/test/README.md

```diff
@@ -0,0 +1,65 @@
+# SDK Codegen Tests
+
+TestSDKCodegen runs the complete set of SDK code generation tests
+against a particular language's code generator. It also verifies
+that the generated code is structurally sound.
+
+The test files live in `pkg/codegen/testing/test/testdata` and
+are registered in `var sdkTests` in `sdk_driver.go`.
+
+An SDK code generation test files consists of a schema and a set of
+expected outputs for each language. Each test is structured as a
+directory that contains that information:
+
+```
+ testdata/
+     my-simple-schema/   # i.e. `simple-enum-schema`
+         schema.(json|yaml)
+         go/
+         python/
+         nodejs/
+         dotnet/
+         ...
+```
+
+The schema is the only piece that *must* be manually authored.
+
+Once the schema has been written, the actual codegen outputs can be
+generated by running the following in `pkg/codegen` directory:
+
+```bash
+$ PULUMI_ACCEPT=true go test ./...
+```
+
+This will rebuild subfolders such as `go/` from scratch and store
+the set of code-generated file names in `go/codegen-manifest.json`.
```

Comment:
We should probably mention how to narrow it down since there are so many tests.
````suggestion
```bash
PULUMI_ACCEPT=true go test ./...
```

This will rebuild subfolders such as `go/` from scratch and store
the set of code-generated file names in `go/codegen-manifest.json`.
To generate the code for a specific directory in testdata,
run the following instead:

```bash
PULUMI_ACCEPT=true go test ./... -run TestGenerate/$dirName
```

````

(Also, I dropped the `$` at the start because you marked this as a bash command. This keeps it easily copy-pasteable.)